drivers/entropy/entropy_native_posix.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2018 Oticon A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Pseudo-random entropy generator for the ARCH_POSIX architecture:
+ * Following the principle of reproducibility of the native_posix board
+ * this entropy device will always generate the same random sequence when
+ * initialized with the same seed
+ */
+
+#include "device.h"
+#include "entropy.h"
+#include "init.h"
+#include <stdlib.h>
+
+static unsigned int seed = 0x5678;
+
+void entropy_native_posix_set_seed(unsigned int seed_i)
+{
+	seed = seed_i;
+}
+
+static int entropy_native_posix_get_entropy(struct device *dev, u8_t *buffer,
```

Comment:
Using `memcpy()` should make this slightly smaller (and a tiny wee little bit faster):

```C
while (length) {
        long int value = random();
        size_t to_copy = min(length, sizeof(value));

        memcpy(buffer, &value, to_copy);
        buffer += to_copy;
        len -= to_copy;
}
```

Reply:
Actually when I wrote that, I was leaning to just take 1 byte from each `random()` call for the shake of simplicity. The amount of time spent generating random bytes being probably quite minimal anyhow.